aiohue/sensors.py

```diff
@@ -0,0 +1,55 @@
+from .api import APIItems
+
+
+class Sensors(APIItems):
+    """Represents Hue Sensors.
+
+    https://developers.meethue.com/documentation/sensors-api
+    """
+
+    def __init__(self, raw, request):
+        super().__init__(raw, request, 'sensors', Sensor)
+
+
+class Sensor:
+    """Represents a Hue sensor."""
+    def __init__(self, id, raw, request):
+        self.id = id
+        self.raw = raw
+        self._request = request
+
+    @property
+    def name(self):
+        return self.raw['name']
+
+    @property
+    def type(self):
+        return self.raw['type']
+
+    @property
+    def modelid(self):
+        return self.raw['modelid']
+
+    @property
+    def manufacturername(self):
+        return self.raw['manufacturername']
+
+    @property
+    def productname(self):
+        return self.raw.get('productname')
+
+    @property
+    def uniqueid(self):
+        return self.raw.get('uniqueid')
```

Comment:
This one is always available as per https://developers.meethue.com/documentation/supported-sensors#generalSensorResource

Reply:
It does say that, and hardware-based items should have it, and if you create a virtual sensor via the API the uniqueid parameter is required, but Hue builds-in a sensor called "Daylight" (it's part of the software of the bridge itself, not a piece of hardware) that does _not_ have this parameter and it can't be added via a REST call. I've griped about it on the Hue forums, but for now it's a clear example where just returning self.raw['uniqueid'] will raise a KeyError for each and every user.

Reply:
okay, let's keep this as `get`.